sklearn/metrics/scorer.py

```diff
@@ -313,12 +322,12 @@ def make_scorer(score_func, greater_is_better=True, needs_proba=False,
 
 # Standard regression scores
 r2_scorer = make_scorer(r2_score)
-mean_squared_error_scorer = make_scorer(mean_squared_error,
-                                        greater_is_better=False)
-mean_absolute_error_scorer = make_scorer(mean_absolute_error,
-                                         greater_is_better=False)
-median_absolute_error_scorer = make_scorer(median_absolute_error,
-                                           greater_is_better=False)
```

Comment:
Neither of the `deprecated()._decorate_*` work because `make_scorer` doesn't return a class nor a function :-/ To get the `__name__` of a class instance you have to do `obj.__class__.__name__`.

I like handling the deprecation stuff as a wrapper instead of extra argument. I made a first proposal in 09cd4c7. It still has a few problems but it would allow you to do:

`deprecated('hello')(make_scorer(median_absolute_error, greater_is_better=False))`

This will however print `Thing _PredictScorer in deprecated...` and I think it will be quite hard to find out the correct name to print in a generic way inside `deprecated()` ... but do we want to cook something that is special to handling `make_scorer`??